cmd/livecheck.rb

```diff
@@ -83,7 +83,9 @@ def livecheck
     # Identify any non-homebrew/core taps in use for current formulae
     non_core_taps = {}
     formulae_to_check.each do |f|
-      non_core_taps[f.tap.name] = true unless f.tap.name == "homebrew/core" || non_core_taps.key?(f.tap.name)
+      unless f.tap.nil?
+        non_core_taps[f.tap.name] = true unless f.tap.name == "homebrew/core" || non_core_taps.key?(f.tap.name)
+      end
```

Comment:
```suggestion
      unless f.tap.nil? || f.tap.name == "homebrew/core" || non_core_taps.key?(f.tap.name)
        non_core_taps[f.tap.name] = true
      end
```

Since this only has one line, it seems like we can potentially tidy it up a bit by moving the other conditions into the outer `unless`.

Reply:
To me, it seems like that whole `unless` line can be simplified to `unless f.tap&.name == "homebrew/core"`

Reply:
Ah, thanks for the reminder. That was the other approach I was going to suggest (the safe navigation operator) but I completely forgot by the time I got to the computer, ahah.

I confirmed that this works fine for me locally (i.e., the safe navigation operator takes the place of `f.tap.nil?` and we're fine to remove the additional `|| non_core_taps.key?(f.tap.name)` condition), so I updated the suggestion above.

Reply:
This change still creates an error since it's accessing the `f.tap.name` at the beginning of the line. Hence I get 
```
Error: undefined method `name' for nil:NilClass
```

Reply:
Ahh, yeah, `f.tap&.name` will be `nil` in this case and `nil` technically doesn't equal `"homebrew/core"`, so it passes through.

Looks like `f.tap.nil?` is the only way to go after all. The current suggestion removes the unnecessary `|| non_core_taps.key?(f.tap.name)`, which lets us maintain the inline conditional. I've tested this locally and it works as expected for a formula without a tap.